stdlib/public/core/Filter.swift

```diff
@@ -140,7 +111,16 @@ public func == <Base : Collection>(
   lhs: LazyFilterIndex<Base>,
   rhs: LazyFilterIndex<Base>
 ) -> Bool {
-  return lhs.base == rhs.base
+  return lhs.baseIndex == rhs.baseIndex
+}
+
+/// Returns `true` iff `lhs` is less than `rhs`.
+@warn_unused_result
+public func < <Base : Collection>(
+  lhs: LazyFilterIndex<Base>,
+  rhs: LazyFilterIndex<Base>
+) -> Bool {
+    return lhs.baseIndex < rhs.baseIndex
 }
```

Comment:
As in the complete set of...

`    @warn_unused_result
    public func <(lhs: Self, rhs: Self) -> Bool
    @warn_unused_result
    public func <=(lhs: Self, rhs: Self) -> Bool
    @warn_unused_result
    public func >=(lhs: Self, rhs: Self) -> Bool
    @warn_unused_result
    public func >(lhs: Self, rhs: Self) -> Bool`